src/Mod/PartDesign/App/FeatureThickness.cpp

```diff
@@ -61,51 +62,111 @@ short Thickness::mustExecute() const
     return DressUp::mustExecute();
 }
 
-App::DocumentObjectExecReturn *Thickness::execute()
+App::DocumentObjectExecReturn* Thickness::execute()
 {
     // Base shape
     Part::TopoShape TopShape;
     try {
         TopShape = getBaseTopoShape();
     }
-    catch (Base::Exception &e) {
+    catch (Base::Exception& e) {
         return new App::DocumentObjectExecReturn(e.what());
     }
 
-    const std::vector<std::string>& subStrings = Base.getSubValues();
+    const std::vector<std::string>& subStrings = Base.getSubValues(true);
 
-    //If no element is selected, then we use a copy of previous feature.
+    // If the base has no subelements listed just return a copy of the base.
     if (subStrings.empty()) {
         //We must set the placement of the feature in case it's empty.
         this->positionByBaseFeature();
         this->Shape.setValue(TopShape);
         return App::DocumentObject::StdReturn;
     }
 
-    /* If the feature was empty at some point, then Placement was set by positionByBaseFeature.
-    *  However makeThickSolid apparently requires the placement to be empty, so we have to clear it*/
+    /* If the feature was ever empty, then Placement was set by positionByBaseFeature.  However
+     * makeThickSolid apparently requires the placement to be empty, so we have to clear it */
     this->Placement.setValue(Base::Placement());
 
-    TopTools_ListOfShape closingFaces;
-
-    for (const auto & it : subStrings) {
-        TopoDS_Face face = TopoDS::Face(TopShape.getSubShape(it.c_str()));
-        closingFaces.Append(face);
+    std::map<int, std::vector<TopoShape>> closeFaces;
+    for ( const auto& it : subStrings) {
+    // for (std::vector<std::string>::const_iterator it = subStrings.begin(); it != subStrings.end();
+         // ++it) {
+        TopoDS_Shape face;
+        try {
+            face = TopShape.getSubShape(it.c_str());
+        }
+        catch (...) {
+        }
+        if (face.IsNull())
+            return new App::DocumentObjectExecReturn(
+                QT_TRANSLATE_NOOP("Exception", "Invalid face reference"));
+        // We found the subelement (face) so let's get it's history index in our shape
+        int index = TopShape.findAncestor(face, TopAbs_SOLID);
+        if (!index) {
+            FC_WARN(getFullName() << ": Ignore non-solid face  " << it);
+            continue;
+        }
+        closeFaces[index].emplace_back(face);
     }
 
     bool reversed = Reversed.getValue();
     bool intersection = Intersection.getValue();
-    double thickness =  (reversed ? -1. : 1. )*Value.getValue();
+    double thickness = (reversed ? -1. : 1.) * Value.getValue();
     double tol = Precision::Confusion();
-    short mode = (short)Mode.getValue();
-    short join = (short)Join.getValue();
+    auto mode = static_cast<short>(Mode.getValue());

```

Comment:
I haven't seen this linter complaint before ("don't use the C type 'short', use int16"), but I think we should ignore it since the function signature of makeElementThickSolid uses short. I could be persuaded though.

Reply:
I had the same thought and left it after changing the C casts.  I think the linter settings (or capabilities or default settings) have changed.

Unless we start changing signatures wholesale in the code base to eliminate the use of short, there will be lots of pointless casts mucking up the code.  So I vote no for now.